lib/probes/http.js

```diff
@@ -260,7 +309,70 @@ function patchServer (module, protocol) {
         ao.bindEmitter(req)
         ao.bindEmitter(res)
 
-        const span = res._ao_http_span = makeSpan(req, settings)
+        const kvpairs = {
+          'Spec': 'ws',
+          'ClientIP': req.socket.remoteAddress,
+          'HTTP-Host': getHost(req),
+          'Port': getPort(req),
+          'Method': req.method,
+          'URL': getPath(req),
+          'Proto': protocol
+        };
+
+        function addKeysToKVPairs () {
+          // now add known keys to the span's kvpairs
+          if (xtraceOptsHash['pd-keys']) {
+            kvpairs['PDKeys'] = xtraceOptsHash['pd-keys'];
+            delete xtraceOptsHash['pd-keys'];
+          }
+          for (const k in xtraceOptsHash) {
+            if (k.startsWith('custom-')) {
+              kvpairs[k] = xtraceOptsHash[k];
+              delete xtraceOptsHash[k];
+            }
+          }
+        }
+
+        // if an x-trace-options header set x-trace-options-response header
+        // if there is an x-trace we need to handle the response as oboe doesn't generate
+        // appopriate responses for those cases.
+        if (xtraceOpts) {
+          const responseParts = [];
+
+          if (!ao.cfg.triggerTraceEnabled) {
+            responseParts.push('trigger-trace=trigger-trace-disabled');
+          } else if (xtrace) {
+            responseParts.push(`trigger-trace=${ttRequested ? 'ignored' : 'not-requested'}`);
+            addKeysToKVPairs();
+          } else {
+            // only add auth key to the response header if an x-trace-options-signature header
+            if (xtraceOptsSig) {
+              responseParts.push(`auth=${settings.authMessage}`);
+            }
+
+            // if the auth status is "good" add other trigger trace response items.
+            if (settings.authStatus <= 0) {
+              // set the options-response header to either trigger-trace status code or not requested
+              responseParts.push(`trigger-trace=${ttRequested ? settings.message : 'not-requested'}`)
+
+              if (ttRequested) {
+                kvpairs.TriggeredTrace = true;
+              }
+
+              addKeysToKVPairs();
+
+              // finally add any ignored keys to the response header
+              const ignored = Object.keys(xtraceOptsHash).join(',');
+              if (ignored) {
+                responseParts.push(`ignored=${ignored}`)
+              }
+            }
+          }
+
+          xtraceOptsResponse = responseParts.join(';');
```

Comment:
if responseParts are empty this returns an empty string, is that falsy?
Asking because it is used later
```
if (optionsResponse) {
      res.setHeader('X-Trace-Options-Response', optionsResponse);
    }
```

Reply:
yes, an empty string is falsey. not an empty array or object, but an empty string is.